crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -1607,8 +1607,8 @@ unsafe impl<'__w, T: Component> WorldQuery for &'__w mut T {
 }
 
 /// SAFETY: access of `&T` is a subset of `&mut T`
-unsafe impl<'__w, T: Component> QueryData for &'__w mut T {
-    type ReadOnly = &'__w T;
+unsafe impl<T: Component> QueryData for &mut T {
+    type ReadOnly = &'static T;
```

Comment:
```suggestion
unsafe impl<SomeComponent: Component> QueryData for &mut SomeComponent {
    type ReadOnly = &'static SomeComponent;
```

We could even go a step further with the above. Output diagnostics:

```
error[E0277]: `query::fetch::test::A` is not valid to request as data in a `Query`
    --> crates\bevy_ecs\src\query\fetch.rs:1414:17
     |
1414 |     let query = Query::<A>::new();
     |                 ^^^^^^^^^^ invalid `Query` data
     |
     = help: the trait `query::fetch::QueryData` is not implemented for `query::fetch::test::A`
     = help: the following other types implement trait `query::fetch::QueryData`:
               &SomeComponent
               &archetype::Archetype
               &mut SomeComponent
               ()
               (F,)
               (F0, F1)
               (F0, F1, F2)
               (F0, F1, F2, F3)
             and 41 others
```